typed-racket-test/unit-tests/subtype-tests.rkt

```diff
@@ -253,9 +253,13 @@
     [FAIL (-pair -String (-lst -Symbol)) (-seq -String)]
     [FAIL (-pair -String (-vec -String)) (-seq -String)]
     [-Null (-seq -String)]
+    [-Null (-mlst -String)]
     [(-mpair -String -Null) (-seq -String)]
+    [(-mpair -String -Null) (-mlst -String)]
```

Comment:
This subtyping is incorrect -- you could mutate the cdr to be a list and then discover that a value of type Null was actually a pair.

Reply:
Did you mean something like this:
```
(: m0 (MPairof String Null))
(define m0 (mcons "hello" '()))
(set-mcdr! m0 (mcons "hello" null))
((inst set-mcdr! String Null) m0 (mcons "Hello" null))
```
Both `set-mcdr! ` expressions are rejected by TR with/without this patch.

My changes is intended to ensure a sequence can be substituted with a null, but not the other way around.

Reply:
What about
```racket
(: m0 (MPairof String Null))
(define m0 (mcons "hello" '()))
((inst set-mcdr! String (MListof String)) m0 (mcons "Hello" null))
```
?
This program should not typecheck, and in master it correctly fails with an error message that `(MPairof String Null)` is not a subtype of `(MListof String)`, and that is how it should be.

`Null` can be a subtype of `(MListof String)`, and it already is, but `MPairof` still needs to be invariant without any `Null` special case.

Reply:
Yes, you are right. M* should always be invariant.